src/main/java/kr/co/antoon/graph/application/GraphScoreSnapshotService.java

```diff
@@ -63,11 +65,43 @@ public GraphScoreResponse graph(Long webtoonId, Period period) {
         return new GraphScoreResponse(scores);
     }
 
+    @Transactional(readOnly = true)
+    public GraphScoreResponse graphByWeekends(Long webtoonId, Period period) {
+        return getGraphScoreResponse(webtoonId, period);
+    }
+
+    @Transactional(readOnly = true)
+    public GraphScoreResponse graphByMonths(Long webtoonId, Period period) {
+        return getGraphScoreResponse(webtoonId, period);
+    }
+
+    @Transactional(readOnly = true)
+    public GraphScoreResponse graphByThreeMonths(Long webtoonId, Period period) {
+        return getGraphScoreResponse(webtoonId, period);
+    }
+
+    private GraphScoreResponse getGraphScoreResponse(Long webtoonId, Period period) {
+        var end = LocalDateTime.now();
+        var start = end.minusDays(period.getDays());
+
+        var graphScoreSnapshots = Stream
+                .iterate(start, date -> date.isBefore(end), date -> date.plusDays(1))
+                .map(date -> graphScoreSnapshotRepository.findTop1ByWebtoonIdAndSnapshotTimeBetweenOrderByCreatedAtDesc(webtoonId, start, end))
```

Comment:
```suggestion
                .map(date -> graphScoreSnapshotRepository.findTopOneByWebtoonIdAndSnapshotTimeBetweenOrderByCreatedAtDesc(webtoonId, start, end))
```

숫자보단 영어로 하는게 더 보기 좋을 것 같습니다.

Reply:
오 findTop1ByWebtoon 이렇게말고 findTopOneByWebtoon 이런 식으로 영어로 해도 되나여? 몰랐네여 감사합니다! 👍